### ISSUE TITLE
method overview widget

### ISSUE THREAD

**Issue by pfitzseb** [JunoLab/atom-julia-client#261]
Add a widget that will show relevant methods (as well as arguments and their types) when the cursor is inside a function call based on
- number of arguments (count commata basically, but not totally trivial)
- known type of supplied arguments (if some argument has already been evaled we know its type and can emulate dispatch)

Also should list relevant keyword arguments.

This should make working with unknown APIs much easier because one doesn't have to look into the docs or the source to figure all that out ([JuDE](https://github.com/jamesdanged/Jude) does something like this already, but not quite as I imagine it).

**Comment by MikeInnes** [comment]
I would really like to have an RStudio-esque doc pane, which can live next to the plot pane. Within that we could have a header which shows "live" information along with anything the user has explicitly requested – that would be really nice.

We could also use the doc pane to give tips on booting and such before the Julia client is booted.

**Comment by pfitzseb** [comment]
Yes, that'd be nice as well (though in general I somewhat dislike panes, but that's not really relevant here :D).

This proposal here was more along the lines of having the method information displayed inline, like [here](http://microsoft.github.io/RTVS-docs/media/intellisense_same_file_functions.png) but with type information and support for multiple methods. Also support for keyword arguments, somewhat like [this](http://microsoft.github.io/RTVS-docs/media/RTVS-REPL-auto-complete-function-parameters.png). [R Tools for VS](http://microsoft.github.io/RTVS-docs/intellisense.html) for some more inspiration.

**Comment by MikeInnes** [comment]
Ah ok. Yeah that could be nice as well, though it might be harder to avoid the usual inline-results issues we have.

It shouldn't be too hard to share the logic anyway, if one is implemented then the other can be.
